src/main/java/com/google/sps/data/AvailabilityTimeSlotGenerator.java

```diff
@@ -119,7 +166,11 @@ static HoursAndMinutes create(int hour, int minute) {
 
       timeSlots.add(
           AvailabilityTimeSlot.create(
-              formattedUTCTime, formattedTime, formattedDate, getSelectedStatus(formattedUTCTime)));
+              formattedUTCTime,
+              formattedTime,
+              formattedDate,
+              getSelectedStatus(formattedUTCTime, userAvailabilityForWeek),
```

Comment:
These `get[x]Status` functions are time costly. Make a HashMap<Instant (startTime), Availability (a member of `userAvailabilityForWeek`)> so you can quickly look up an instant in `userAvailabilityForWeek`.
On the other hand, `userAvailabilityForWeek` won't be very large so l don't know even know if this matters. @jalexanderqed @acmcarther ?

Reply:
Done (refactored to use HashMap).